platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/TestReportGenerator.java

```diff
@@ -16,15 +16,32 @@
 
 package org.gradle.api.internal.tasks.testing;
 
-import org.gradle.internal.operations.BuildOperationExecutor;
-import org.gradle.internal.operations.BuildOperationRunner;
 import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
 
 import java.nio.file.Path;
+import java.util.List;
 
+/**
+ * Shared interface between legacy/JVM and generic test report generators.
+ */
 @NullMarked
 public interface TestReportGenerator {
-    boolean hasResults();
+    /**
+     * A test report generator that does nothing.
+     */
+    TestReportGenerator NO_OP = resultsDirectories -> null;
```

Comment:
❔ Is this actually used by production code, or is it just a test implementation?  If so, it probably shouldn't live here.